packages/design-tokens/scripts/style-dictionary.js

```diff
@@ -0,0 +1,495 @@
+  transformer: token => {
+    if (isPx(token.value)) {
+      const baseFontSize = 16
+      const floatValue = parseFloat(token.value.replace('px', ''))
+
+      if (isNaN(floatValue)) {
+        return token.value
+      }
+
+      if (floatValue === 0) {
+        return '0'
+      }
+
+      if (token.name.includes('lineHeight')) {
+        return `${floatValue / baseFontSize}`
+      }
+
+      return `${floatValue / baseFontSize}rem`
+    }
+    return token.value
+  },
+})
+
+//-----
+
+// ts output
+StyleDictionary.registerTransform({
+  name: 'attribute/typescript',
+  type: 'attribute',
+  transformer: token => {
+    return {
+      typescript: {
+        // these transforms will need to match the ones you use for typescript
+        // or you can "chain" the transforms and use token.name and token.value
+        // like scss and less transforms do.
+        name: token.path.slice(1).join('.'),
+        value: token.value,
+      },
+    }
+  },
+})
+
+// css output
+StyleDictionary.registerTransform({
+  name: 'attribute/css',
+  type: 'attribute',
+  transformer: token => {
+    const tokenName = token.path.slice(1).join('-')
+    return {
+      css: {
+        name: `--${tokenName}`,
+        value: token.value,
+      },
+    }
+  },
+})
+
+// transform: composite typography to shorthands
+StyleDictionary.registerTransform({
+  name: 'typography/shorthand',
+  type: 'value',
+  transitive: true,
+  matcher: token => token.type === 'typography',
+  transformer: token => {
+    const {value} = token
+
+    // if lineHeight has value, include in shorthand
+    if (value.lineHeight) {
+      return `${value.fontWeight} ${value.fontSize}/${value.lineHeight} ${value.fontFamily}`
+    }
+
+    return `${value.fontWeight} ${value.fontSize} ${value.fontFamily}`
+  },
+})
+
+// REGISTER THE CUSTOM TRANFORM GROUPS
+
+StyleDictionary.registerTransformGroup({
+  name: 'css',
+  transforms: ['name/css', 'pxToRem', 'typography/shorthand'],
+})
+
+StyleDictionary.registerTransformGroup({
+  name: 'scss',
+  transforms: ['name/scss', 'pxToRem', 'typography/shorthand'],
+})
+
+// REGISTER A CUSTOM FORMAT
+
+// wrap mobile tokens in media query
+StyleDictionary.registerFormat({
+  name: 'css/touch-target-mobile',
+  formatter({dictionary, file, options}) {
+    const {outputReferences} = options
+    return `${fileHeader({file})}
+    @media (pointer: coarse) { :root {\n${formattedVariables({
+      format: 'css',
+      dictionary,
+      outputReferences,
+    })}\n}}\n`
+  },
+})
+
+// wrap desktop tokens in media query
+StyleDictionary.registerFormat({
+  name: 'css/touch-target-desktop',
+  formatter({dictionary, file, options}) {
+    const {outputReferences} = options
+    return `${fileHeader({file})}
+    @media (pointer: fine) { :root {\n${formattedVariables({
+      format: 'css',
+      dictionary,
+      outputReferences,
+    })}\n}}\n`
+  },
+})
+
+StyleDictionary.registerFormat({
+  name: 'custom/format/custom-media',
```

Comment:
Can you use this: https://github.com/primer/primitives/blob/main/src/formats/cssCustomMedia.ts ?